### DISCUSSION TITLE
Running multiple reusable actions in succession

### DISCUSSION THREAD

**Discussion by leo-petrucci** [community/community#26730] [Actions]
<p>I’m currently experimenting with reusable actions. In my head I was going to separate each part of my action into reusable actions, then call them whenever needed on a step-by-step basis.</p>
<p>This is what I came up with:</p>
<pre><code class="lang-auto">name: Opened Pull Request

on:
  pull_request:
    branches:
      - master
      - staging
    # Allows you to run this workflow manually from the Actions tab
    workflow_dispatch:

jobs:
  pr:
    steps:
      - name: Build and Test
        uses: creativiii/commentcarp-api/.github/workflows/action.build-and-test.yml@staging
        secrets:
          APP_SECRET: "${{ secrets.APP_SECRET }}"

      - name: Compare coverage
        uses: creativiii/commentcarp-api/.github/workflows/action.compare-coverage.yml@staging

</code></pre>
<p>But trying to run this errors out with:</p>
<pre><code class="lang-auto">Error : .github#L1
reusable workflows should be referenced at the top-level `jobs.*.uses' key, not within steps
</code></pre>
<p>So I’m confused how to get multiple reusable workflows in a single action.</p>
<p>I can get it running like this, but they they won’t be run in the order I want them to.</p>
<pre><code class="lang-auto">jobs:
  build:
    uses: creativiii/commentcarp-api/.github/workflows/action.build-and-test.yml@staging
    secrets:
      APP_SECRET: "${{ secrets.APP_SECRET }}"

  compare:
    uses: creativiii/commentcarp-api/.github/workflows/action.compare-coverage.yml@staging

</code></pre>
<p>Any tips?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>What you’re using there are <em>reusable workflows</em>, though it sounds like <em>composite actions</em> might be a better fit for what you want to do: <a href="https://docs.github.com/en/actions/creating-actions/creating-a-composite-action" class="inline-onebox">Creating a composite action - GitHub Docs</a></p>
<p>Alternatively, if you really want to use reusable workflows, you could <a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idneeds">create a <code>needs</code> relationship</a> between the jobs. Transferring data will be harder that way, though.</p>

**Comment by leo-petrucci** [comment]
<p>Oops, I assumed the two were one and the same. I’ll try composite actions, thank you!</p>
